spec/stateful/job_control.py

```diff
@@ -266,6 +266,30 @@ def suspend_status(sh):
   expect_prompt(sh)
 
 
+@register(skip_shells=['zsh'])
+def no_spurious_tty_take(sh):
+  'A background job getting stopped by SIGTTIN should not disrupt foreground processes'
+  expect_prompt(sh)
+
+  sh.sendline('cat &')
+  time.sleep(0.1)  # seems necessary
+
+  # background cat should have been stopped by SIGTTIN immediately, but we don't
+  # hear about it from wait() until the foreground job has been started because
+  # the shell was blocked in readline when the signal fired.
```

Comment:
BTW `set -b`  in POSIX is a getting to give  the notifications immediately

Maybe we should implement that as it does make it a bit easier to reason about the tests

Reply:
Oh cool. I didn't know about that. Would be nice to add